tsdb/chunks/head_chunks_test.go

```diff
@@ -248,23 +257,42 @@ func TestChunkDiskMapper_Truncate(t *testing.T) {
 // This test exposes https://github.com/prometheus/prometheus/issues/7412 where the truncation
 // simply deleted all empty files instead of stopping once it encountered a non-empty file.
 func TestChunkDiskMapper_Truncate_PreservesFileSequence(t *testing.T) {
-	hrw := testChunkDiskMapper(t)
+	hrw := createChunkDiskMapper(t, "")
 	defer func() {
 		require.NoError(t, hrw.Close())
 	}()
-
 	timeRange := 0
+
+	var callbackWg sync.WaitGroup
 	addChunk := func() {
+		t.Helper()
+
+		callbackWg.Add(1)
```

Comment:
Similar to what you did in another test with `awaitCb := make(chan struct{})`, can we do the same here and get rid of this shared `callbackWg`?

Reply:
I replaced it everywhere in 575ff12b8